langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java

```diff
@@ -19,6 +19,11 @@
 import dev.langchain4j.model.chat.listener.ChatModelRequestContext;
```

Comment:
#2120 is still not resolved. It returns the response in plain text format, which causes an error when the response is attempted to be parsed as JSON.

Reply:
Hi @Gikicio , thank you for looking into this!
So both tools and json schema are properly sent to the LLM, but the problem is now with response parsing?
I will try to dig into it this during the week, but any help is highly appreciated 🙏

Reply:
## First path: Without Tools Enabled

The first approach involves generating the response via the method `dev.langchain4j.model.openai.OpenAiChatModel#chat`. The following steps occur:

- The method `Response<AiMessage> response = generate(...)` is called with several parameters:
  - `request.messages()`
  - `request.toolSpecifications()`
  - `null`
  - `getOrDefault(toOpenAiResponseFormat(request.responseFormat(), strictJsonSchema), this.responseFormat)`

In this scenario, the response is correctly set within the `generate` method. The `ResponseFormat` has the correct type (`JSON_SCHEMA`), and the `jsonSchema` itself is properly included. The OpenAI model is then called with the correct parameters, resulting in expected behavior.

## Second path: With Tools Enabled

The second approach involves the use of tools and proceeds via the `DefaultAiServices` class. The `generate` method is called with two parameters at line 280:

```java
response = context.chatModel.generate(messages, toolSpecifications);
```

This call is then forwarded to an overloaded version of `generate` that takes different parameters, specifically involving `this.responseFormat`. In this case, the `jsonSchema` is not set in the bean, and it is expected to be auto-deduced, as in the first scenario. However, the schema that gets into the `generate` method (`dev.langchain4j.model.openai.OpenAiChatModel#generate(...)`) is of type `JSON_SCHEMA`, but the `jsonSchema` itself is `null` because it wasn't set manually.

## Proposed Solution

To solve this issue, it is recommended that the schema be passed as in the first scenario to ensure consistency. This would involve dynamically adding the `jsonSchema` during the tool-enabled process, just as it is done in the first scenario.

Reply:
@Gikicio thank you for looking into it!
There was a bug indeed, I have fixed it and added a `AbstractAiServiceIT.should_execute_tool_then_return_structured_output` test to make sure it works properly